clients/src/main/java/org/apache/kafka/clients/admin/NewPartitionReassignment.java

```diff
@@ -28,17 +28,13 @@
  * A new partition reassignment, which can be applied via {@link AdminClient#alterPartitionReassignments(Map, AlterPartitionReassignmentsOptions)}.
  */
 public class NewPartitionReassignment {
-    private final List<Integer> targetBrokers;
+    private final List<Integer> targetReplicas;
 
-    public static Optional<NewPartitionReassignment> of(Integer... brokers) {
```

Comment:
I am not sure about removing this method. `of` was a nice helper that returns an Optional. To use the Admin API, we now need to do the verbose:
```
Optional.of(new NewPartitionReassignment(Arrays.asList(1, 2, 3)))
```
versus
```
NewPartitionReassignment.of(1,2,3)
```
Note that we want an optional because the Admin API accepts a map of optionals. A `None` value denotes a reassignment cancellation.
```

    /**
     * Change the reassignments for one or more partitions.
     * Providing an empty Optional (e.g via {@link Optional#empty()}) will <bold>revert</bold> the reassignment for the associated partition.
     */
    AlterPartitionReassignmentsResult alterPartitionReassignments(
        Map<TopicPartition, Optional<NewPartitionReassignment>> reassignments,
        AlterPartitionReassignmentsOptions options);
```

If NPEs are a concern, can we just check for those and throw an IllegalArgumentException?

Reply:
@stanislavkozlovski That method was only used in tests. I created a helper method in that class.